client/network/src/service.rs

```diff
@@ -542,23 +559,169 @@ impl<B: BlockT + 'static, H: ExHashT> NetworkService<B, H> {
 		&self.local_peer_id
 	}
 
-	/// Writes a message on an open notifications channel. Has no effect if the notifications
-	/// channel with this protocol name is closed.
+	/// Appends a notification to the buffer of pending outgoing notifications with the given peer.
+	/// Has no effect if the notifications channel with this protocol name is not open.
+	///
+	/// If the buffer of pending outgoing notifications with that peer is full, the notification
+	/// is silently dropped and the connection to the remote will start being shut down. This
+	/// happens if you call this method at a higher rate than the rate at which the peer processes
+	/// these notifications, or if the available network bandwidth is too low.
```

Comment:
What you say is correct, but I make two assumptions here:

- The libp2p task that contains the `Receiver` is light enough that the only reason why it would be too slow is that the remote is too slow or bandwidth is too low.
- Since the channel is quite large (512 elements), it takes a lot of time for the sender to actually fill this channel. Even if the CPU is busy, it is very unlikely that this channel fills up before we had to chance to start processing its content.

Additionally, thanks to the `notifications_queue_size` metric, we can empirically see today that we almost never reach 512 elements in the queue.

Reply:
Ok, if you feel confident about doing this, no objections from my side.